src/Tribe/Adjacent_Events.php

```diff
@@ -189,74 +192,89 @@ public function get_closest_event_where( $where_sql ) {
 	 * Get the prev/next post for a given event. Ordered by start date instead of ID.
 	 *
 	 * @since 4.6.12
+	 * @since TBD Cache the query results.
 	 *
 	 * @param string  $mode Either 'next' or 'previous'.
 	 *
-	 * @return null|WP_Post
+	 * @return null|WP_Post The closest Event post object, or `null` if no post was found.
 	 */
 	public function get_closest_event( $mode = 'next' ) {
-		$post_obj = get_post( $this->current_event_id );
-
-		if ( 'previous' === $mode ) {
-			$order      = 'DESC';
-			$direction  = '<';
-		} else {
-			$order      = 'ASC';
-			$direction  = '>';
-			$mode       = 'next';
+		if ( empty( $this->current_event_id ) ) {
+			return null;
 		}
 
-		$args = [
-			'posts_per_page' => 1,
-			'post__not_in'   => [ $this->current_event_id ],
-			'meta_query'     => [
-				[
-					'key'     => '_EventStartDate',
-					'value'   => $post_obj->_EventStartDate,
-					'type'    => 'DATETIME',
-					'compare' => $direction,
-				],
-				[
-					'key'     => '_EventHideFromUpcoming',
-					'compare' => 'NOT EXISTS',
-				],
-				'relation'    => 'AND',
-			],
-		];
-
-		$events_orm = tribe_events();
-
-		/**
-		 * Allows the query arguments used when retrieving the next/previous event link
-		 * to be modified.
-		 *
-		 * @since 4.6.12
-		 *
-		 * @param array   $args
-		 * @param WP_Post $post_obj
-		 */
-		$args = (array) apply_filters( "tribe_events_get_{$mode}_event_link", $args, $post_obj );
-
-		$events_orm->order_by( 'event_date', $order );
-		$events_orm->by_args( $args );
-		$query = $events_orm->get_query();
-
-		// Make sure we are not including same datetime events
-		add_filter( 'posts_where', [ $this, 'get_closest_event_where' ] );
-
-		// Fetch the posts
-		$query->get_posts();
-
-		// Remove this filter right after fetching the events
-		remove_filter( 'posts_where', [ $this, 'get_closest_event_where' ] );
+		$cache     = tribe_cache();
+		$cache_key = 'tec_events_closest_event_' . $this->current_event_id . '_' . $mode;
+		// The cached value will be the post ID, or `null`, to avoid pre-fetch issues.
+		$cached = $cache->get( $cache_key, Cache_Listener::TRIGGER_SAVE_POST, false );
+		$event = $cached;
+		if ( ! empty( $cached ) ) {
+			// If not empty, it should be a valid event post ID.
+			$event = get_post( $cached );
+			if ( ! ( $event instanceof WP_Post && $event->post_type === TEC::POSTTYPE ) ) {
+				$event = false;
+			}
+		} elseif ( $cached !== null ) {
```

Comment:
```
		} elseif ( $cached !== null ) {
```
After the 
```
                if ( ! empty( $cached ) ) {
```
seems odd, it is targeting similar things but I can see a hole in the logic if it is `false` or something else. Is this intentional?

Reply:
The expected value should be either a post ID or `null`.
If the `$cached` value is not empty, then it should either be a valid event, or return `false`.
If the `$cached` value is empty, the only valid allowed value is `null`, and that is covered by the second check.

If `$cached` is `0`, then the `$cached !== null` check would fail and the function would return `false`.

Am I missing something?

Reply:
It's a good idea to add a test for this, in any case.